functional_tests.go

```diff
@@ -4709,6 +4713,7 @@ func testUserMetadataCopyingWrapper(c *minio.Client) {
 
 	expectedHeaders := make(http.Header)
 	expectedHeaders.Set("x-amz-meta-notmyheader", "notmyvalue")
+	expectedHeaders.Set("content-type", "application/javascript")
```

Comment:
While this is a good fix.. There is a lot of simplification missing @donatello . It was significantly harder to add these parts of the code. For example if you look at the getProps() filters out x-amz-meta- only and ignores all headers..

Reply:
We should perhaps take a stab at simplifying the metadata handling in ComposeObject and avoid it being interwoven.. perhaps you can think about this further.

Reply:
Yes, it is complicated. The difficulty seems to be due to the lack of a simple way to extract  from  `ObjectInfo` the set of values that can be set via headers. I will give it some thought.